src/rdkafka_conf.h

```diff
@@ -289,6 +289,14 @@ struct rd_kafka_conf_s {
                         rd_kafka_t *rk,
                         const char *oauthbearer_config,
                         void *opaque);
+                struct {
+                        char *method;
+                        char *token_endpoint_url;
+                        char *client_id;
+                        char *client_secret;
+                        char *scope;
+                        char **extensions;
```

Comment:
`extensions` is a i _RK_C_STR property, which expects a `char *`, but here it is a `char **`.
What you need to do is have this as a `char *extensions_str` and then in conf_finalize() translate the CSV to an rd_list_t that populates `rd_list_t extensions;`, and also performs key=value format checking.

Reply:
You can probably move the oauthbearer_token_refresh_cb into this oauthbearer struct (and remove the oauthbearer_ prefix from that field).